pilot/pkg/bootstrap/server.go

```diff
@@ -647,9 +660,55 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 		mcpClient := sink.NewClient(cl, sinkOptions)
 		configz.Register(mcpClient)
 		clients = append(clients, mcpClient)
-
 		conns = append(conns, conn)
 		configStores = append(configStores, mcpController)
+
+		// create MCP SyntheticServiceEntryController
+		if len(args.SSEAddrs) > 0 {
+			s.discoveryOptions = &coredatamodel.DiscoveryOptions{
+				DomainSuffix: args.Config.ControllerOptions.DomainSuffix,
+			}
+			s.mcpDiscovery = coredatamodel.NewMCPDiscovery(s.discoveryOptions)
+			s.mcpDiscovery.InitNetworkLookup(s.meshNetworks)
+
+		}
+		for _, addr := range args.SSEAddrs {
+			s.incrementalMcpOptions = &coredatamodel.Options{
```

Comment:
for example this block can move to a NewSomething ? 
Is there any difference between an MCP source for configs and one for endpoints ? Why ? 
( I mean - besides the list of resources that are watched ).

Reply:
I believe the separation in MCP source is intentional for scalability reasons, since it is not needed to have as many MCP source server that forward configs as endpoints. @nmittler perhaps could clarify.